src/gpu/intel/jit/conv/plan.cpp

```diff
@@ -1243,6 +1243,17 @@ std::string conv_plan_t::str() const {
     return jit::add_indent("conv_plan", oss.str());
 }
 
+type_t get_accumulation_type(
+        fma_kind_t fma_kind, const type_t &a, const type_t &b) {
+    if (a.is_int()) return type_t::s32();
+    if (a.is_f64()) return type_t::f64();
+    if (fma_kind == fma_kind_t::mad && a.is_f16() && b.is_f16()) {
+        // XXX: f16 must use f32 accumulator according to documentation.
+        return type_t::f16();
```

Comment:
Let's just go ahead and fix this, even if it causes regressions. We already made this change for many other implementations and as regressions are expected, leaving this implementation as exempt due to regressions doesn't seem sensical.

Reply:
Hmm, I don't think this is a good idea. We have a lot of CNN workloads and intentionally introducing a regression without giving a heads-up to our customers (at least major ones) does not look right to me. For example we have a lot of depthwise layers regressing quite a bit after switching f16 to f32 accumulator so I suggest to be more careful here.

I created a Jira to work on that: https://jira.devtools.intel.com/browse/MFDNN-13271